chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts

```diff
@@ -13,25 +13,34 @@ import { IReconnectChatContext } from "../../reconnectchatpanestateful/interface
 import { IReconnectChatOptionalParams } from "../../reconnectchatpanestateful/interfaces/IReconnectChatOptionalParams";
 import { LiveChatWidgetActionType } from "../../../contexts/common/LiveChatWidgetActionType";
 import { TelemetryHelper } from "../../../common/telemetry/TelemetryHelper";
+import { handleAuthentication, removeAuthTokenProvider } from "./authHelper";
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const getChatReconnectContext = async (chatSDK: any, reconnectId?: string) => {
-    try {
-        if (reconnectId) {
-            const chatReconnectOptionalParams: IReconnectChatOptionalParams = {
-                reconnectId: reconnectId
-            };
-            return await chatSDK?.getChatReconnectContext(chatReconnectOptionalParams);
-        } else {
-            return await chatSDK?.getChatReconnectContext();
-        }
-    } catch (ex) {
-        TelemetryHelper.logSDKEvent(LogLevel.ERROR, {
-            Event: TelemetryEvent.GetChatReconnectContextSDKCallFailed,
-            ExceptionDetails: {
-                exception: ex
+const getChatReconnectContext = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, isReconnectEnabled?: boolean, reconnectId?: string) => {
+    if (isReconnectEnabled) {
+        try {
+            if (reconnectId) {
+                const chatReconnectOptionalParams: IReconnectChatOptionalParams = {
+                    reconnectId: reconnectId
+                };
+                return await chatSDK?.getChatReconnectContext(chatReconnectOptionalParams);
+            } else {
+                // set auth token to chat sdk to get reconnectId for auth chat
+                await handleAuthentication(chatSDK, chatConfig, getAuthToken);
+                const reconnectChatContext = await chatSDK?.getChatReconnectContext();
+                // remove auth token after reconnectId is fetched
+                // this will be reset later at start chat
+                removeAuthTokenProvider(chatSDK);
+                return reconnectChatContext;
             }
-        });
+        } catch (ex) {
+            TelemetryHelper.logSDKEvent(LogLevel.ERROR, {
+                Event: TelemetryEvent.GetChatReconnectContextSDKCallFailed,
```

Comment:
Please ensure this is published to as lcw:error when adding script code

Reply:
will do